contracts/Removal.sol

```diff
@@ -23,22 +23,29 @@ contract Removal is
 {
   using SafeMathUpgradeable for uint256;
 
-  struct Vintage {
-    address supplier;
-    uint16 vintage;
-    // todo: location
-    // todo: methodology
-    // todo: supplier name
-  }
-
   struct BatchMintRemovalsData {
     address marketAddress;
     bool list;
   }
 
-  mapping(uint256 => Vintage) private _vintages;
+  uint256 constant _BITS_PER_BYTE = 8;
+  uint256 constant _ADDRESS_FIELD_LENGTH = 20;
+  uint256 constant _ADDRESS_OFFSET = 12;
+  uint256 constant _PARCEL_ID_FIELD_LENGTH = 5;
+  uint256 constant _PARCEL_ID_OFFSET = 7;
+  uint256 constant _VINTAGE_FIELD_LENGTH = 2;
+  uint256 constant _VINTAGE_OFFSET = 5;
+  uint256 constant _COUNTRY_CODE_FIELD_LENGTH = 2;
+  uint256 constant _COUNTRY_CODE_OFFSET = 3;
+  uint256 constant _STATE_CODE_FIELD_LENGTH = 2;
+  uint256 constant _STATE_CODE_OFFSET = 1;
+  uint256 constant _METHODOLOGY_DATA_FIELD_LENGTH = 1;
+  uint256 constant _METHODOLOGY_DATA_OFFSET = 0;
```

Comment:
since these are all internal constants, it's chill to declare them all here without incurring additional storage gas costs (according to the internet) because their instances will be replaced with the constant value by the optimizer.  Just much better for readability like this.